src/dune_engine/dep.ml

```diff
@@ -255,19 +251,24 @@ module Facts = struct
     in
     Fact.Files.group fact_files paths
 
-  let dirs t =
-    Map.fold t ~init:Path.Set.empty ~f:(fun fact acc ->
-        match (fact : Fact.t) with
-        | Nothing | File _ -> acc
-        | File_selector (_, ps) | Alias ps ->
-          Path.Set.union acc (Path.Map.keys ps.dirs |> Path.Set.of_list))
-
-  let parent_dirs t =
-    Map.fold t ~init:Path.Set.empty ~f:(fun fact acc ->
+  let necessary_dirs_for_sandboxing t =
+    Map.fold t ~init:Path.Build.Set.empty ~f:(fun fact acc ->
         match (fact : Fact.t) with
         | Nothing -> acc
-        | File (p, _) -> Path.Set.add acc (Path.parent_exn p)
-        | File_selector (_, ps) | Alias ps -> Path.Set.union acc ps.parent_dirs)
+        | File (p, _) -> (
+          match Path.as_in_build_dir p with
+          | None -> acc
```

Comment:
Same here I suppose (i.e. we should raise on seeing a source path here).

Reply:
Done